migration/identity-migration/src/main/java/io/camunda/migration/identity/transformer/AuthorizationTransformer.java

```diff
@@ -0,0 +1,95 @@
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.migration.identity.transformer;
+
+import io.camunda.migration.identity.dto.Role.Permission;
+import io.camunda.service.AuthorizationServices.PatchAuthorizationRequest;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.PermissionAction;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class AuthorizationTransformer {
+
+  static final List<AuthorizationResourceType> RESOURCE_TYPES =
+      Arrays.asList(
+          AuthorizationResourceType.PROCESS_DEFINITION,
+          AuthorizationResourceType.DECISION_DEFINITION,
+          AuthorizationResourceType.DECISION_REQUIREMENTS_DEFINITION);
+
+  public static List<PatchAuthorizationRequest> transform(
+      final long ownerKey, final List<Permission> oldPermissions) {
+    final var groupedPermissions =
+        oldPermissions.stream()
+            .flatMap(AuthorizationTransformer::transformToAuthorizations)
+            .collect(
+                Collectors.groupingBy(
+                    ResourceTypePermissionTypeResourceId::resourceType,
+                    Collectors.groupingBy(
+                        ResourceTypePermissionTypeResourceId::permissionType,
+                        Collectors.mapping(
+                            ResourceTypePermissionTypeResourceId::resourceId,
+                            Collectors.toSet()))));
+    return groupedPermissions.entrySet().stream()
+        .map(entry -> createPatchAuthorizationRequest(ownerKey, entry))
+        .toList();
+  }
+
+  private static PatchAuthorizationRequest createPatchAuthorizationRequest(
+      final long ownerKey,
+      final Entry<AuthorizationResourceType, Map<PermissionType, Set<String>>> entry) {
+    return new PatchAuthorizationRequest(
+        ownerKey, PermissionAction.ADD, entry.getKey(), entry.getValue());
+  }
+
+  private static Stream<ResourceTypePermissionTypeResourceId> transformToAuthorizations(
+      final Permission permission) {
+
+    final List<String> applications = List.of("operate", "tasklist");
+    return applications.stream()
+        .filter(a -> permission.apiName().toLowerCase().contains(a))
+        .flatMap(
+            a -> {
+              final var applicationAccess =
+                  Stream.of(
+                      new ResourceTypePermissionTypeResourceId(
+                          AuthorizationResourceType.APPLICATION, PermissionType.ACCESS, a));
+              if (permission.definition().toLowerCase().contains("read")) {
```

Comment:
:question: Why `contains` and not `equals` here?

Reply:
I have used contains to cover both `read:*` and `read` permissions.